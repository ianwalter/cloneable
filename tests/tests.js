const { test } = require('@ianwalter/bff')
const cloneable = require('..')

test('object', ({ expect }) => {
  const thing = {
    message: 'Hello!',
    yellMessage () {
      return this.message.toUpperCase()
    },
    song: {
      index: 3,
      date: new Date(),
      getLyrics () {
        return 'Darling its bout time to break free of'
      }
    }
  }
  expect(cloneable(thing)).toMatchSnapshot({
    song: {
      date: expect.any(Date)
    }
  })
})

test('array', ({ expect }) => {
  const thing = [
    {
      title: 'Bury Us',
      updates: [9, 10]
    },
    [
      {
        regex: /(naked|famous)/,
        search () {
          this.results = []
        },
        results: [
          {
            title: 'Stay Forever',
            add () {
              this.count++
            },
            otherActions: [
              function remove () {}
            ]
          }
        ]
      }
    ]
  ]
  expect(cloneable(thing)).toMatchSnapshot()
})
