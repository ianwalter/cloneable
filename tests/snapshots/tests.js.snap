// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`array 1`] = `
Array [
  Object {
    "title": "Bury Us",
    "updates": Array [
      9,
      10,
    ],
  },
  Array [
    Object {
      "regex": Object {
        "lastIndex": 0,
      },
      "results": Array [
        Object {
          "otherActions": Array [],
          "title": "Stay Forever",
        },
      ],
    },
  ],
]
`;

exports[`cloneable 1`] = `
Object {
  "message": "Hello!",
  "song": Object {
    "index": 3,
  },
}
`;

exports[`object 1`] = `
Object {
  "message": "Hello!",
  "song": Object {
    "date": Any<Date>,
    "index": 3,
  },
}
`;
